url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+		panel.add(new JLabel("Number of ENTER: "));	
+		panel.add(new JLabel("Number of Ask: "));	
+		panel.add(new JLabel("Number of Bid: "));
+		panel.add(new JLabel("Number of TRADE: "));	
+		panel.add(new JLabel("Number of AMEND: "));
+		panel.add(new JLabel("Number of DELETE: "));
+
+		panel2.add(stockName);
+		panel2.add(stockDate);
+		panel2.add(stockTime);
+		panel2.add(LinesRead);
+		panel2.add(enterLines);
+		panel2.add(askLines);
+		panel2.add(bidLines);
+		panel2.add(tradeLines);
+		panel2.add(amendLines);
+		panel2.add(deleteLines);
+
+		stockName.setText(overviewResult.get(0));
+		stockDate.setText(overviewResult.get(1));
+		stockTime.setText(overviewResult.get(2));
+		LinesRead.setText(overviewResult.get(3));
+		enterLines.setText(overviewResult.get(4));
+		askLines.setText(overviewResult.get(5));
+		bidLines.setText(overviewResult.get(6));
+		tradeLines.setText(overviewResult.get(7));
+		amendLines.setText(overviewResult.get(8));
+		deleteLines.setText(overviewResult.get(6));
+
+		analysispanel.add(panel);
+		analysispanel.add(panel2);
+
+		toppanel.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.gridy = 0;	
+		analysispanel.setBorder(BorderFactory.createEtchedBorder(EtchedBorder.LOWERED));
+
+		toppanel.add(analysispanel,c);
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 2;
+		c.gridx = 1;
+		c.gridy = 0;
+		toppanel.add(new JPanel(),c);
+
+		return toppanel;
+	}
+	private FinalOrderbookTable ordertable;
+
+	private JPanel orderbookPanel() {
+		JPanel panel = new JPanel();
+		ordertable = new FinalOrderbookTable();
+		JTable buybook = new JTable();
+		buybook.setModel(ordertable);
+
+		try {
+			ResultSet set = myDB.getResultSet("SELECT * FROM old_trade_list;");
+			while(set.next()){
+				long bidID = set.getLong(12);
+				long askID = set.getLong(13);
+				double tmpPrice = set.getDouble(6);
+				int tmpVol = set.getInt(7);
+				Time tmpTime = set.getTime(3);
+				Object [] fakedata1 = {bidID,askID, "$ " + tmpPrice, tmpVol, tmpTime};
+				ordertable.addElement(fakedata1);
+			};
+			set.close();
+		}catch (SQLException e) {
+			System.out.println("In Mainmenu/insertTables : " + e);
+		}
+
+
+		Dimension d = new Dimension (650,500);
+
+		JScrollPane scrollTable = new JScrollPane(buybook);
+		buybook.setFillsViewportHeight(true);
+		scrollTable.setPreferredSize(d);
+		scrollTable.setMaximumSize(d);	
+
+		panel.add(scrollTable);
+		return panel;
+	}
+
+	private JPanel graphPanel(){
+
+		JPanel panel = new JPanel();
+		JPanel graphpanel = new JPanel();
+
+		panel.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+
+		//create new graph and data set
+		LineGraph returntimegraph = new LineGraph("All accounts from " + title.substring(13));
+		//add plots to the graph
+		try {
+			ResultSet set = myDB.getResultSet("SELECT * FROM all_list;");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *set*. The resource is referenced at lines 262, 263, 265, 266, 267, 275, 276. The resource is closed at line 281. There are other execution paths that do not contain closure statements, for example, when ResultSet.next throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).